pkg/scorecard_test.go

```diff
@@ -187,3 +191,128 @@ func TestRunScorecard(t *testing.T) {
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    ScorecardResult
+		wantErr bool
+	}{
+		{
+			name: "empty commits repos should return repo details but no checks",
+			args: args{
+				uri:       "github.com/ossf/scorecard",
+				commitSHA: "1a17bb812fb2ac23e9d09e86e122f8b67563aed7",
+				probes:    []string{"fuzzedWithOSSFuzz"},
+			},
+			want: ScorecardResult{
+				Repo: RepoInfo{
+					Name:      "github.com/ossf/scorecard",
+					CommitSHA: "1a17bb812fb2ac23e9d09e86e122f8b67563aed7",
+				},
+				RawResults: checker.RawResults{
+					Metadata: checker.MetadataData{
+						Metadata: map[string]string{
+							"repository.defaultBranch": "main",
+							"repository.host":          "github.com",
+							"repository.name":          "ossf/scorecard",
+							"repository.sha1":          "1a17bb812fb2ac23e9d09e86e122f8b67563aed7",
+							"repository.uri":           "github.com/ossf/scorecard",
+						},
+					},
+				},
+				Scorecard: ScorecardInfo{
+					CommitSHA: "unknown",
+				},
+				Findings: []finding.Finding{
+					{
+						Probe:   "fuzzedWithOSSFuzz",
+						Message: "no OSSFuzz integration found",
+						Remediation: &probe.Remediation{
+							Text: fmt.Sprintf("%s%s%s\n%s%s",
+								"Follow the steps in ",
+								"https://github.com/google/oss-fuzz ",
+								"to integrate fuzzing for your project.",
+								"Over time, try to add fuzzing for more ",
+								"functionalities of your project."),
+							Markdown: fmt.Sprintf("%s%s%s\n%s%s",
+								"Follow the steps in [https://github.com",
+								"/google/oss-fuzz](https://github.com/google/oss-fuzz) ",
+								"to integrate fuzzing for your project.",
+								"Over time, try to add fuzzing for more ",
+								"functionalities of your project."),
+							Effort: 3,
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Wrong probe",
+			args: args{
+				uri:       "github.com/ossf/scorecard",
+				commitSHA: "1a17bb812fb2ac23e9d09e86e122f8b67563aed7",
+				probes:    []string{"nonExistentProbe"},
+			},
+			want:    ScorecardResult{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ctrl := gomock.NewController(t)
+			mockRepoClient := mockrepo.NewMockRepoClient(ctrl)
+			repo := mockrepo.NewMockRepo(ctrl)
+
+			repo.EXPECT().URI().Return(tt.args.uri).AnyTimes()
+			repo.EXPECT().Host().Return("github.com").AnyTimes()
+
+			mockRepoClient.EXPECT().InitRepo(repo, tt.args.commitSHA, 0).Return(nil)
+
+			mockRepoClient.EXPECT().Close().DoAndReturn(func() error {
+				return nil
+			})
+
+			mockRepoClient.EXPECT().ListCommits().DoAndReturn(func() ([]clients.Commit, error) {
+				if tt.args.commitSHA == "" {
+					return []clients.Commit{}, nil
+				}
+				return []clients.Commit{
+					{
+						SHA: tt.args.commitSHA,
+					},
+				}, nil
+			})
+			mockRepoClient.EXPECT().GetDefaultBranchName().Return("main", nil).AnyTimes()
+			defer ctrl.Finish()
```

Comment:
`ctrl.Finish()` isn't needed anymore, but if we have it, at least keep it right after the line which defines it.
> New in go1.14+, if you are passing a *testing.T into NewController function you no longer need to call ctrl.Finish() in your test methods.

Reply:
Removed in https://github.com/ossf/scorecard/pull/3688/commits/876b07149c711bd9d92a540532ca7b447c393539